lib/sequelize.js

```diff
@@ -577,11 +577,11 @@ class Sequelize {
               return this.connectionManager.releaseConnection(connection);
             }
           });
-      }), retryOptions)).finally(() => {
+      }).finally(() => {
         if (this.test._trackRunningQueries) {
           this.test._runningQueries--;
         }
-      });
+      }), retryOptions));
```

Comment:
If query is retried for any reason (could be `options.retry`) the `_runningQueries` count will become negative. I suspect this is why it works now.

Please verify if `_runningQueries` count is ever negative in your local tests

Reply:
oof. you are right. I verified locally and you were right, my bad. 
I went ahead and played around with it a little more and figured out that it makes sense though to increase and decrease the counter on every try because there is the edge-case of looking at hooks (especially when I think of the pre and post query hooks we talked about earlier) and therefore, it would still be marked as running although it's about to attempt the next try. What do you think?